dotCMS/src/main/java/com/dotmarketing/webdav/DotWebdavHelper.java

```diff
@@ -156,49 +156,17 @@ public boolean isAutoPub(String path){
 		}
 	}
 
-	public User authorizePrincipal(String username, String passwd)	throws DotSecurityException, NoSuchUserException, DotDataException {
-		User _user;
+	public User authorizePrincipal(String username, String password)	throws DotSecurityException, NoSuchUserException, DotDataException {
 
-		boolean useEmailAsLogin = true;
-		Company comp = com.dotmarketing.cms.factories.PublicCompanyFactory.getDefaultCompany();
-		if (comp.getAuthType().equals(Company.AUTH_TYPE_ID)) {
-			useEmailAsLogin = false;
-		}
-		try {
-			if (PRE_AUTHENTICATOR != null && !PRE_AUTHENTICATOR.equals("")) {
-				Authenticator authenticator;
-				authenticator = (Authenticator) new com.dotcms.repackage.bsh.Interpreter().eval("new " + PRE_AUTHENTICATOR + "()");
-				if (useEmailAsLogin) {
-					authenticator.authenticateByEmailAddress(comp.getCompanyId(), username, passwd);
-				} else {
-					authenticator.authenticateByUserId(comp.getCompanyId(), username, passwd);
-				}
-			}
-		}catch (AuthException ae) {
-			Logger.debug(this, "Username : " + username + " failed to login", ae);
-			throw new DotSecurityException(ae.getMessage(),ae);
-		}catch (Exception e) {
-			Logger.error(this, e.getMessage(), e);
-			throw new DotSecurityException(e.getMessage(),e);
-		}
-		UserAPI userAPI=APILocator.getUserAPI();
-		if (comp.getAuthType().equals(Company.AUTH_TYPE_ID)) {
-			_user = userAPI.loadUserById(username,userAPI.getSystemUser(),false);
-		} else {
-			_user = userAPI.loadByUserByEmail(username, userAPI.getSystemUser(), false);
-		}
 
-        if (_user == null) {
-            throw new DotSecurityException("The user was returned NULL");
-        }
+            try {
+                APILocator.getLoginServiceAPI().doLogin(username, password) ;
+            } catch (com.liferay.portal.NoSuchUserException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)